src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/Zaimanhua.kt

```diff
@@ -233,37 +241,48 @@ class Zaimanhua : HttpSource(), ConfigurableSource {
         apiHeaders,
     )
 
+    private fun genreApiUrl(): HttpUrl.Builder =
+        "$apiUrl/comic/filter/list".toHttpUrl().newBuilder()
+            .addQueryParameter("size", DEFAULT_PAGE_SIZE.toString())
+
     override fun popularMangaParse(response: Response): MangasPage = latestUpdatesParse(response)
 
     // Search
     private fun searchApiUrl(): HttpUrl.Builder =
         "$apiUrl/search/index".toHttpUrl().newBuilder().addQueryParameter("source", "0")
-            .addQueryParameter("size", "20")
+            .addQueryParameter("size", DEFAULT_PAGE_SIZE.toString())
 
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
         val ranking = filters.filterIsInstance<RankingGroup>().firstOrNull()
-        val url = if (query.isEmpty() && ranking != null) {
-            rankApiUrl().apply {
-                ranking.state.filterIsInstance<QueryFilter>().forEach {
-                    it.addQuery(this)
-                }
+        val genres = filters.filterIsInstance<GenreGroup>().firstOrNull()
```

Comment:
```suggestion
        val genres = filters.firstInstanceOrNull<GenreGroup>()
```
 (import from `keiyoushi.utils`)